src/qt/walletmodel.cpp

```diff
@@ -756,3 +779,296 @@ bool WalletModel::getDefaultWalletRbf() const
 {
     return fWalletRbf;
 }
+
+bool WalletModel::nameAvailable(const QString &name)
+{
+    JSONRPCRequest jsonRequest;
+    UniValue params (UniValue::VOBJ);
+    UniValue res, array, isExpired;
+
+    const std::string strName = name.toStdString();
+    params.push_back (Pair("name", strName));
+    jsonRequest.strMethod = "name_show";
+    jsonRequest.params = params;
+    jsonRequest.fHelp = false;
+
+    try {
+        res = tableRPC.execute(jsonRequest);
+
+    } catch (const UniValue& e) {
+        return true;
+    }
+
+    isExpired = find_value( res, "expired");
+    if(isExpired.get_bool())
+      return true;
+
+    return false;
+}
+
+NameNewReturn WalletModel::nameNew(const QString &name)
+{
+    std::string strName = name.toStdString ();
+    std::string data;
+
+    JSONRPCRequest jsonRequest;
+    UniValue params(UniValue::VOBJ);
+    std::vector<UniValue> values;
+    UniValue res, txid, rand;
+
+    NameNewReturn retval;
+
+    params.push_back (Pair("name", strName));
+
+    jsonRequest.strMethod = std::string("name_new");
+    jsonRequest.params = params;
+    jsonRequest.fHelp = false;
+
+    try {
+        res = name_new (jsonRequest);
+    } catch (const UniValue& e) {
+        UniValue message = find_value( e, "message");
+        std::string errorStr = message.get_str();
+        LogPrintf ("nameNew error: %s\n", errorStr.c_str());
+        retval.err_msg = errorStr;
+        retval.ok = false;
+        return retval;
+    }
+
+    retval.ok = true;
+
+    values = res.getValues ();
+    txid = values[0];
+    rand = values[1];
```

Comment:
If you move the declarations here, you can also change them to be const references, and possibly use `auto`.